beacon-chain/blockchain/chain_info.go

```diff
@@ -69,6 +67,8 @@ type HeadFetcher interface {
 type ForkFetcher interface {
```

Comment:
These changes reduce the interface surface area for various functions that don't need a full `ChainInfoFetcher`, but just a subset of methods. Composition helps us a lot, as we can implement a smaller mock that fits the interface when needed